torch/nn/utils/rnn.py

```diff
@@ -360,6 +360,8 @@ def pad_sequence(sequences, batch_first=False, padding_value=0.0):
 
     # assuming trailing dimensions and type of all the Tensors
     # in sequences are same and fetching those from sequences[0]
+    if isinstance(sequences, torch.Tensor):
```

Comment:
This has to be `Tensor` as TorchScript doesn't support `Iterable`.

Reply:
But here we want to convert any Iterable into a tuple before calling into the c++ impl. Because the C++ impl expects a tuple?

Reply:
@kshitij12345 is it possible to support all iterables outside of TorchScript with appropriate `torch.jit.is_scripting()` / `torch.jit.is_tracing()` gating? Could support just tuple / Tensor for the TorchScript case.

Reply:
`is_scripting` and `is_tracing` works. Thanks @albanD @jbschlosser.